phoenix-core/src/it/java/org/apache/phoenix/end2end/DeleteIT.java

```diff
@@ -817,7 +838,59 @@ private void testDeleteCount(boolean autoCommit, Integer limit) throws Exception
         }
 
     }
-    
+
+    private void testDeleteCountWithIndex(boolean autoCommit, boolean scn) throws Exception {
+        String tableName = generateUniqueName();
+        String tableDDL = "CREATE TABLE IF NOT EXISTS " + tableName + " (pk1 DECIMAL NOT NULL, v1 VARCHAR CONSTRAINT PK PRIMARY KEY (pk1))";
+        String indexName = generateUniqueName();
+        String indexDDL = "CREATE INDEX IF NOT EXISTS " + indexName + " ON " + tableName + "(v1)";
+        int numRecords = 1010;
```

Comment:
do we need so many rows to test the timestamp behavior? How much time do these tests add to the test suite run? Seems like you could get the same effect in the scn case with 2 rows, using the edge manager to force a two millisecond time increment between the two upserts.

Reply:
If you do go that way, please remember to reset() the EnvironmentEdgeManager in a finally block

Reply:
Each test takes about 5 to 10 seconds even with 1010 rows. I needed to see the same timestamp is used across multiple batches so I need to have several batches of rows. I have not injected an EnvironmentEdge and so no need to reset.

Reply:
OK, that's fine. Thanks!